vicodyn/include/cocaine/vicodyn/balancer/utils.hpp

```diff
@@ -17,7 +26,10 @@ auto choose_random_if(Iterator begin, Iterator end, std::size_t size, Predicate
     if(chosen.empty()) {
         return end;
     }
-    return chosen[rand() % chosen.size()];
+    if (chosen.size() <= RAND_MAX + 1) {
+        return chosen[std::rand() % chosen.size()];
```

Comment:
looks quite strange, what is the need for runtime algo choosing?

Reply:
this is premature optimization and should be avoided

Reply:
IMO

Reply:
Will be fixed because std::mt19937 will be used, thank you.